src/main/java/org/folio/circulation/domain/notice/schedule/DueDateNotRealTimeScheduledNoticeHandler.java

```diff
@@ -27,32 +31,37 @@
 import io.vertx.core.json.JsonObject;
 
 public class DueDateNotRealTimeScheduledNoticeHandler {
-  public static DueDateNotRealTimeScheduledNoticeHandler using(Clients clients, DateTime systemTime) {
+
+  public static DueDateNotRealTimeScheduledNoticeHandler using(Clients clients, DateTime systemTime, EventPublisher eventPublisher) {
```

Comment:
We can add `PubSubPublishingService` to the `Clients` class, so that `EventPublisher` can be created explicitly here instead of carry them from the outer code.

The `PubSubPublishingService` should be refactored to the following:
```java
public class PubSubPublishingService {
  private static final Logger logger = LoggerFactory.getLogger(PubSubPublishingService.class);

  // we need only tenant id here.
  private final String tenant;
  private final PubsubClient pubSubClient;

  public PubSubPublishingService(RoutingContext routingContext) {
    this(new WebContext(routingContext));
  }

// new constructor that uses circ WebContext instead vertx RoutingContext
  public PubSubPublishingService(WebContext context) {
    this.tenant = context.getTenantId();
    this.pubSubClient = new PubsubClient(context.getOkapiLocation(), context.getTenantId(),
      context.getOkapiToken());
  }
// rest is the same
}
```

Then In the Clients just create the client similarly to others.

Then we can create EventPublisher directly in the class that uses it (I assume it is notice service).